V1/Interfaces/Services/IEncryptionService.cs

```diff
@@ -0,0 +1,27 @@
+using System.Threading.Tasks;
+// TODO: All references to WalletContract below will change to reference the "new" version
+using CoviIDApiCore.V1.DTOs.Wallet;
+// TODO: All references to CovidTestCredentialParameters below will change to reference the "new" version
+using CoviIDApiCore.V1.DTOs.Credentials;
+
+namespace CoviIDApiCore.V1.Interfaces.Services
+{
+    public interface IEncryptionService
+    {
+        Task<string> GenerateEncryptedSecretKey();
+
+        // These methods only really care about the mobile-number related
+        // fields, but we accept the full wallet type for consistency.
+        //
+        // NOTE: These don't take in an encryptedSecretKey, because they are
+        // encrypted using the serverKey only.
+        Task<WalletContract> EncryptWallet(WalletContract plainTextWallet);
+        Task<WalletContract> DecryptWallet(WalletContract encryptedWallet);
+
+        Task<WalletContract> EncryptWalletDetails(WalletContract plainTextWallet, string encryptedSecretKey);
```

Comment:
It would be easier to take a model in of type T, so any model would be able to be en- and decrypted.
This would avoid creating multiple methods to encrypt and decrypt the model

Reply:
But then you would not know which fields have to be encrypted vs which not because you don't have context of the model content. Ie, the Id should not be encrypted but the FirstName should.

Reply:
Yeah I thought about this - it's definitely possible. We could do some logic using `typeof(T)` and branching. My reasoning here was that you shouldn't be allowed to pass _any_ type in - just the ones which make sense.

So if I passed in something weird like
```
encryptionService.Encrypt<Foo>(myFoo)
```
Should it throw an error?


What I really wanted was to use a union type, but I don't think C# has those yet?